src/Esri.ArcGISRuntime.Toolkit/Android/UI/Controls/Compass.Android.cs

```diff
@@ -0,0 +1,221 @@
+        public Compass(Context context, IAttributeSet attr) : base(context, attr)
+        {
+            Initialize();
+        }
+        
+        /// <inheritdoc />
+        protected override LayoutParams GenerateDefaultLayoutParams()
+        {
+            var size = (int)CalculateScreenDimension((float)DefaultSize);
+            return new LayoutParams(size, size);
+        }
+
+        private void Initialize()
+        {
+            var size = (int)CalculateScreenDimension((float)DefaultSize);
+            _northArrow = new NorthArrowShape(this.Context) { Size = size };
+            AddView(_northArrow);
+            UpdateCompassRotation(false);
+            _northArrow.Click += NorthArrow_Click;
+        }
+
+        private void NorthArrow_Click(object sender, EventArgs e)
+        {
+            ResetRotation();
+        }
+
+        private void UpdateCompassRotation(bool transition)
+        {
+            SetVisibility(!AutoHide || Heading != 0, transition);
+            _northArrow.Rotation = -(float)this.Heading;
+        }
+
+        /// <inheritdoc />
+        protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
+        {
+            base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
+
+            // Initialize dimensions of root layout
+            MeasureChild(_northArrow, MeasureSpec.MakeMeasureSpec(MeasureSpec.GetSize(widthMeasureSpec), MeasureSpecMode.AtMost), MeasureSpec.MakeMeasureSpec(MeasureSpec.GetSize(heightMeasureSpec), MeasureSpecMode.AtMost));
+
+            // Calculate the ideal width and height for the view
+            var desiredWidth = PaddingLeft + PaddingRight + _northArrow.MeasuredWidth;
+            var desiredHeight = PaddingTop + PaddingBottom + _northArrow.MeasuredHeight;
+
+            // Get the width and height of the view given any width and height constraints indicated by the width and height spec values
+            var width = ResolveSize(desiredWidth, widthMeasureSpec);
+            var height = ResolveSize(desiredHeight, heightMeasureSpec);
+            SetMeasuredDimension(width, height);
+        }
+
+        /// <inheritdoc />
+        protected override void OnLayout(bool changed, int l, int t, int r, int b)
+        {
+            // Forward layout call to the root layout
+            _northArrow.Layout(PaddingLeft, PaddingTop, _northArrow.MeasuredWidth + PaddingLeft, _northArrow.MeasuredHeight + PaddingBottom);
+        }
+
+        private void SetVisibility(bool isVisible, bool animate = true)
+        {
+            if (animate)
+            {
+                if(isVisible)
+                {
+                    if (_fadeInAnimation != null)
+                        return; //Already fading in
+                    if (_fadeOutAnimation != null)
+                    {
+                        _fadeOutAnimation.Cancel();
+                        _fadeOutAnimation = null;
```

Comment:
Does the `WithEnd` action fire if the animation is cancelled?  If so, the animation would be set to null twice.  Not a big deal, but it would be unnecessary.